## Xa7 QTL regions:
#### Xa7QTL was mapped to a genetic interval of 118.5 kb (0.21-cM) between SSR markers GDSSR02 and RM20593 [1]
* RM20593
  * Forward: 5’-AAGGTACACTTGCTCTGACGGTAGCIRGSP-1.0: 25 bp from chr06:28,082,317..28,082,341
  * Reverse: 5’-AGACCTCAGTGGCAAATCCTACGIRGSP-1.0: 23 bp from chr06:28,082,610..28,082,632
* GDSSR02
  * Forward: 5’-TGCCCACCGTCGAACTCGTGG, IRGSP-1.0: 21 bp from chr06:27,963,796..27,963,816 
  * Reverse: 5’-AGCTAGCAATTCGCATGATTGCIRGSP-1.0: 22 bp from chr06:27,963,981..27,964,002
* GDSSR02 to RM20593: 118,836 bp from chr06:27,963,796.. 28,082,632
* [1] Shen Chen, et al. High-resolution mapping and gene prediction of Xanthomonas Oryzae pv. Oryzae resistance gene Xa7. Molecular Breeding;2008;22(3):433-441.

```bash
>RM20593F_Xa7QTL-118R
AAGGTACACTTGCTCTGACGGTAGC
>RM20593R_Xa7QTL-118R
AGACCTCAGTGGCAAATCCTACG

>GDSSR02F_Xa7QTL-118L
TGCCCACCGTCGAACTCGTGG
>GDSSR02R_Xa7QTL-118L
AGCTAGCAATTCGCATGATTGC

```

#### Xa7QTL was mapped to a genetic interval of 200-kb between SSR marker RM20576 and STS marker MY4 [2]
* RM20576 
  * Forward: 5’-CTGTTGCTAGCTTACACGAATTGCIRGSP-1.0: 24 bp from chr06:27,890,361..27,890,384
  * Reverse: 5’-CCGGTAGTACGTCAGCTACTATGCIRGSP-1.0: 24 bp from chr06:27,890,627..27,890,650
* MY4
  * Forward: 5’-CTTCCTTTCCAGGCTTTC IRGSP-1.0: 18 bp from chr06:28,090,186..28,090,203
  * Reverse: 5’-TAGAATTTGCATCATCCCIRGSP-1.0: 18 bp from chr06:28,090,356..28,090,373
* RM20576 to MY4: 200,012  bp from chr06:27,890,361..28,090,373
* [2] Zhang, Yuchen, et al. Identification and molecular mapping of the rice bacterial blight resistance gene allelic to Xa7 from an elite restorer line Zhenhui 084. Eur J Plant Pathol;2009;125:235–244. 

```bash
>RM20576F_Xa7QTL-200L
CTGTTGCTAGCTTACACGAATTGC
>RM20576R_Xa7QTL-200L
CCGGTAGTACGTCAGCTACTATGC

>MY4F_Xa7QTL-200R
CTTCCTTTCCAGGCTTTC
>MY4R_Xa7QTL-200R
TAGAATTTGCATCATCCC

```

#### Fine mapping of Xa7 by Dr. Bing Yang Lab at 01/20/2015 indicates the gene is within a 56 kb region between two markers M5 and M5-56k:
```
>M5_Xa7QTL_BY
CGATCTTACTGGCTCTGCAACTCTGTATTGCATGCTAAATCCGTTATGATTTACTACTAATGATGGATGTACTGATACATAGTACTGATCGATGAGCTAGCTAGGGTTTTGGATGGGTCGGAAGGTGAGAAAGAGGAGGAAAGAAGAGAGAATCTATTATATTATTAAAGGAATAGAAAAAGAAGCCTCCACGTTCGCTCTCACGGCCTAGAAATTCTCACATTAATCGGAGAAAAGAAAAAGCAGAGTCCATATAGAAATACAATTTAGAAATAGCTGAAATTCGGAATTATAAAATAAGGAATATTAGAAGAGGAGACTAGAGTCCATATGGAAATACAATTTAGAAATAGTTGAAATTCAGAATTAAAAAATAAGAAATATTAGAAGAGGAGACTAGAGTCCATATAGAAATATAATTAGGAAATAACTGAAATTCGGAATTAAAAATAAGGAATATTAGAAGTAGAGTATAGAGTCCATATAAAAATATAATTAGGAAATAACTGAAATTAGGAATTAATAATAAGGAATATTAGAGATAGAGTATAGAGTCCATATAAAAATACAATTAGTAAATAACTGTAATTCGGAATTAAAAATAAGGAATATTAGAGGTAGAGTATAGAGTCCATATAGAAATACAATTAGAAAATAACTGTAATTCGGAATTAAAAATAAGGAATATTAGAGGTAGAGTATAGAGTCCATATAAAAATACAATTAGGAAATAACTGAAATTCGGAATTAAAAATAAGGAATATTAGAGGTAGAGTATAGAGTCCATATAGAAATACAATTAGAAAATAATAAAAATTCGGAATTAAAAATTTGATATTAAAAATAATTAATAACTAACACGTATATATAATACAATATAAATATTACACATTAGTAGTTTTACAAAATTTAAAATTATATTGTCATTTTAATAAATTTGAATAATACATTGAGAAAACATATATGCTATTACATGAGAGAAAATATAATGATGCTAGCCGCGCAATATGCACGGGCCACTATGCTAGTTGAAGATGATTAGAAGAAATCTAATCAATTTCAGAGCATTACTGTATTCATCTTTCAGTGTATATTCGTCTGGGTAGTGCTGATTGTGCCATTATAATTGACGGGGTCGTACTTCGTACGGACGAATCGACACAGACATGCATGATTGTCTTTTAGTACTTAGAATTAGTAGCAAAGTCGGTGGGACAAC
>M5_56k_Xa7QTL_BY
AGAAAACGGATCAGACAAGCATTCTTTGTTTAAACGTGTAGACATCTCCTTATGTTACTCCTCTTGTACATGTGGGTTGTTTCTTTGGGGTGAACTTAGGAATCTTCCTAGCAAATGCCCTTGCTGTAAATTTACAGTACAGGATCAGGTGGGTACACCTCTACACATTTCAGAAGCTTTGGCGGGGAGCTTAGCTCTGCATTGGTGTTACTCGCAACTTTTGGAGGCAACCGCTGCATTAACAAACAGCACCAAAACCACATTTGAACGCTGCAGCATGTAATGTTGCAGTTTACTCGCACGGGGACAAACGATATCTATTTATTTGTACGCAGTAGAGTGTTAGGAGGAGGTGACAGAGCGAGCAAAACAACAAAAAGGACGTGTGTGTATATAGCTCCAAGTTGGGATTTTATGAGTAGTACAGAAACCATGGATCTTACTGGGTAGTAACCCCAGCTTTTGTCGTCCCAAAAAGATGAGATGTCGATCCCTCGATCTCGCTCTCTCCTCTTTACGTGCAAAAATTAAAAAGGACATGGCCATTTTTGTGGCTTCATTTTCTCGACATTTAAATGGAAACATTGATACAGATTTCATGTTGCTGTTGATTCAGGATTAGATTATCCTTTCCGTTTTTTTACACCTGCCGGTGCATTTTCCGATGGGTGTAGCTCCGAGCTTCGTTAACTAACGGCTCGCCCACACTCTCCTCCCAAAAACGATGTATAGATCACTCTGCACTGTACATGCTCTCCAACACGTTGACCACCCTTCTCCCGAGGCCGCTCGCCTCGCCGGTGCACCACCGGTGAGCGCTCCGTCCGAGCCGGTCGGCTCATGACCATCCTGACCAATCTTTTACATACCGCAAAGGCCGTCATCGTTGCCGGGGACCAAGACGATGAACAGTGGGAGATCGCCGGCCGGCTGGGACCATGGCGGCGGTGGCGATGTACTCTCTCTGTTTCATAAAAAACAAACCTAGGATCAGATATGACATATTCTAATATAATAAATCTGAATATATATATATATATATGTAGATTCGTAATATATTCGGCACTAGTAGGTTGGTTTTTTATGGGACACTCGGGCGGCCAGCCCAATCATGTGAGGAAATCGTACCGAAAAGACCAGGCTGCCCCTGGTACGTGTGTGTAGATAGGCGGAATCTAGCGAGGGGGCCAATGCTTTGCAAATGCAATGCCATTGCTGCCGGCACGGGCGAGCATATGTCTCTGACGCGACGACACGTCCGTCCCTCCCTTCCCCTGTCCCGACCTGTCGCCTTGTATACGCTCGTACTAGGTACAGTAGTACGTACGGTTGCGATACCTCCGTAGCTACGGTGAACACGCGAAGTAATACTCCGTATCCCCGTAGACATAACCGTATTCTCTAGTAGTACTACTACTACAGAAGAAACTGGGCCCTAAAATCTACTACTGCCATAGCACTCACAAAACGTAGGGGGCCAATGTAATGCAGTAGTTAATCATACTGTAGCAGTTTGTTTGAGTAAGGTGCATGAGCAGTCCTTAATTTTATACGGGCATGTTATTTAGATACCTAATATGACATCCCGGCTCAAGGCTTAATAGAATTAATAGAATACTCATATCAACAAGTTGCAACTAATTTTTCGGAAGCCGGTCTCGAAAGAACTCTTAGGTTAAGTGTGCTTGGCCTGAAGCAATTTTGGGATGGGTGACCGACTGGGAAATTCTTCCTGTGTGTGCACG

```
* M5_M5-56K in rice japonica is chr6:27,965,437..28,022,337, Length=56,901 bp.
* M5_M5-56K in rice indica is chr6:29,683,670..29,828,319, Length=144,650 bp.
* M5_M5-56K in rice indica contains an artifact insert region in chr6:29,704,162..29,783,684, Length=79,523 bp, which was possibly originated from contaminated bacterial sequence. After removing this artifact insert, M5_M5-56K in rice indica becomes 66,477 bp.
* Seven protein-coding genes are included in this region. They can be used as anchors. They are: OS06G0673700, OS06G0674000, OS06G0674100, OS06G0674400, OS06G0674800, OS06G0675200, OS06G0675300.
* Among them, there are two pairs of overlapping genes: "OS06G0673700, OS06G0674000", and "OS06G0675200, OS06G0675300".

## Rice cultivars that are Xoo resistant
* CX134 IRBB7 Indica Note: [6] Xa7
* IRIS 313-10605 DV86 Aus/boro  Note: [2][4] Xa7;Xa5
* CX369 IRBB62 Indica Note: [5] Xa4;Xa7;Xa21
* CX126 IRBB60 Indica Note: [3][4] Xa4;xa5;xa13;Xa21
* CX44 IR58025B Indica Note: [3][4] Xa4;xa5;xa13;Xa21

## REFERENCE:
1. Schatz MC, et al. Whole genome de novo assemblies of three divergent strains of rice, Oryza sativa, document novel gene space of aus and indica. Genome Biol. 2014;15(11):506.
2. Sakai H, et al. Construction of pseudomolecule sequences of the aus rice cultivar Kasalath for comparative genomics of Asian cultivated rice. DNA Res. 2014 Aug;21(4):397-405.


#### Blastn
```bash
# Blastn parameters
BLASTDB=${TRegGA_DIR}/reference/rice_DJ123/OsdDNA
DB_REFSEQ=OsdDNA
QUERY=Xa7MarkerDNA.fa
QUERYNAME=Xa7Marker
QUERYSHORT=Xa7Markershort.fa
QUERYNAMESHORT=Xa7Markershort

# OUTFMT is the tabular output format for BlastDB query
OUTFMT='6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore gaps qcovs qcovhsp'
EVALUE="1e-100"
PERC_IDENTITY=92

# Word length of initial exact match, blastn default is 11, blastn-short default is 7.
# blastn -db ${BLASTDB} -query ${QUERY} -evalue ${EVALUE} -perc_identity=${PERC_IDENTITY} > blastn.${QUERYNAME}-on-${DB_REFSEQ}.aln
# blastn -db ${BLASTDB} -query ${QUERY} -evalue ${EVALUE} -perc_identity=${PERC_IDENTITY} -outfmt "${OUTFMT}" >\
# blastn.${QUERYNAME}-on-${DB_REFSEQ}.table

## If adjust for short sequences
# -task "blastn-short", -word_size 7, -evalue 1000, -dust no
# blastn -db ${BLASTDB} -query ${QUERYSHORT} -task "blastn-short" -word_size 11 -evalue 10 -perc_identity=100 -outfmt "${OUTFMT}" >\
# blastn.${QUERYNAMESHORT}-on-${DB_REFSEQ}.table
# grep "Chr6" blastn.${QUERYNAMESHORT}-on-${DB_REFSEQ}.table > blastn.${QUERYNAMESHORT}-on-${DB_REFSEQ}_Chr6.table
```

#### Haplovars DID NOT improve the contig N50 whatsoever, SAD. 
#### Alternatively, use Aus DJ123 as reference for TRegGA.
* 1. DJ123 reference sequence from http://schatzlab.cshl.edu/data/rice/
* 2. Marker sequence for XA7 M5-56K region in DJ123
```bash
# Download rice indica IR64 [1] contigs and scaffolds and annotations
cd ${TRegGA_DIR}/reference
mkdir rice_IR64
cd rice_IR64
./xgetseq
cd -

# Download rice aus DJ123 [1] contigs and scaffolds and annotations
cd ${TRegGA_DIR}/reference
mkdir rice_DJ123
cd rice_DJ123
./xgetseq
cd -

# Blastn the XA7Marker against rice_DJ123 database
mkdir -p ${TRegGA_DIR}/run/DJ123
cd ${TRegGA_DIR}/run/DJ123
\cp ${TRegGA_DIR}/doc/M5_56K/XA7MarkerDNA.fa .
BLASTDB=${TRegGA_DIR}/reference/rice_DJ123/OsdDNA
DB_REFSEQ=OsdDNA

QUERY=XA7MarkerDNA.fa
QUERYNAME=XA7Marker
# OUTFMT is the tabular output format for BlastDB query
OUTFMT='6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore gaps qcovs qcovhsp'
EVALUE="1e-10"
PERC_IDENTITY=90

blastn -db ${BLASTDB} -query ${QUERY} -evalue ${EVALUE} -perc_identity=${PERC_IDENTITY} > blastn.${QUERYNAME}-on-${DB_REFSEQ}.aln
blastn -db ${BLASTDB} -query ${QUERY} -evalue ${EVALUE} -perc_identity=${PERC_IDENTITY} -outfmt "${OUTFMT}" > blastn.${QUERYNAME}-on-${DB_REFSEQ}.table

* 1. M5-56K in rice aus DJ123 is scaffold_10:197,410..199,218, Length=1,809 bp. Retrieve scaffold_10
query=scaffold_10
blastdbcmd -db ${BLASTDB} -entry "lcl|${query}" -dbtype nucl -outfmt "${OUTFMT}" -out ${DB_REFSEQ}-${query}.fa
    sed -i -e "s/>lcl|/>/" ${DB_REFSEQ}-${query}.fa
# python getTarget.py ${i}-${query}.fa ${i}-${query}_GAP_${qGapStart}_${qGapEnd} ${qGapStart} ${qGapEnd}
# sed -i "s/^>${query}/>${i}-${query}_GAP_${qGapStart}_${qGapEnd}/" ${i}-${query}_GAP_${qGapStart}_${qGapEnd}.fasta


# Download rice aus Kasalath [2] reference genome
cd ${TRegGA_DIR}/reference
mkdir rice_aus
cd rice_aus
./xgetseq
cd -


# Blastn the XA7Marker against rice_aus database
mkdir -p ${TRegGA_DIR}/run/Kasalath
cd ${TRegGA_DIR}/run/Kasalath
\cp ${TRegGA_DIR}/doc/M5_56K/XA7MarkerDNA.fa . 
BLASTDB=${TRegGA_DIR}/reference/rice_japonica/OsjDNA
DB_REFSEQ=OsjDNA
BLASTDB=${TRegGA_DIR}/reference/rice_indica/OsiDNA
DB_REFSEQ=OsiDNA
BLASTDB=${TRegGA_DIR}/reference/rice_aus/OsaDNA
DB_REFSEQ=OsaDNA

QUERY=Xa7MarkerDNA.fa
QUERYNAME=Xa7Marker
QUERYSHORT=Xa7Markershort.fa
QUERYNAMESHORT=Xa7Markershort

# OUTFMT is the tabular output format for BlastDB query
OUTFMT='6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore gaps qcovs qcovhsp'
EVALUE="1e-100"
PERC_IDENTITY=92
# Word length of initial exact match, blastn default is 11, blastn-short default is 7. 
blastn -db ${BLASTDB} -query ${QUERY} -evalue ${EVALUE} -perc_identity=${PERC_IDENTITY} > blastn.${QUERYNAME}-on-${DB_REFSEQ}.aln
blastn -db ${BLASTDB} -query ${QUERY} -evalue ${EVALUE} -perc_identity=${PERC_IDENTITY} -outfmt "${OUTFMT}" > blastn.${QUERYNAME}-on-${DB_REFSEQ}.table

blastn -db ${BLASTDB} -query ${QUERYSHORT} -task "blastn-short" -word_size 11 -evalue 10 -perc_identity=100 -outfmt "${OUTFMT}" > blastn.${QUERYNAMESHORT}-on-${DB_REFSEQ}.table
grep "Chr6" blastn.${QUERYNAMESHORT}-on-${DB_REFSEQ}.table > blastn.${QUERYNAMESHORT}-on-${DB_REFSEQ}_Chr6.table

# blastn.Xa7Markershort-on-OsjDNA_Chr6.table
M5_Xa7QTL_BY    OsjChr6 100.00  1219    0       0       1       1219    27965437        27966655        0.0      2252   0       100     100
M5_56k_Xa7QTL_BY        OsjChr6 100.00  1745    0       0       1       1745    28020593        28022337        0.0      3223   0       100     100

RM20593F_Xa7QTL-118R    OsjChr6 100.00  25      0       0       1       25      28082317        28082341        3e-06   50.1    0       100     100
GDSSR02R_Xa7QTL-118L    OsjChr6 100.00  22      0       0       1       22      27964002        27963981        1e-04   44.1    0       100     100
RM20576F_Xa7QTL-200L    OsjChr6 100.00  24      0       0       1       24      27890361        27890384        1e-05   48.1    0       100     100
RM20576R_Xa7QTL-200L    OsjChr6 100.00  24      0       0       1       24      27890650        27890627        1e-05   48.1    0       100     100
MY4F_Xa7QTL-200R        OsjChr6 100.00  18      0       0       1       18      28090186        28090203        0.014   36.2    0       100     100
MY4R_Xa7QTL-200R        OsjChr6 100.00  18      0       0       1       18      28090373        28090356        0.014   36.2    0       100     100

# blastn.Xa7Markershort-on-OsiDNA_Chr6.table
M5_Xa7QTL_BY    OsiChr6 95.09   977     41      7       245     1219    29683915        29684886        0.0      1531   7       88      80
M5_56k_Xa7QTL_BY        OsiChr6 98.67   1430    11      2       1       1430    29826585        29828006        0.0      2529   8       99      82
M5_56k_Xa7QTL_BY        OsiChr6 93.07   361     12      2       1384    1744    29828006        29828353        3e-144    516   13      99      21

RM20593R_Xa7QTL-118R    OsiChr6 100.00  23      0       0       1       23      29902696        29902674        4e-05   46.1    0       100     100
GDSSR02R_Xa7QTL-118L    OsiChr6 100.00  22      0       0       1       22      29681265        29681244        1e-04   44.1    0       100     100
RM20576F_Xa7QTL-200L    OsiChr6 100.00  24      0       0       1       24      29607908        29607931        1e-05   48.1    0       100     100
RM20576R_Xa7QTL-200L    OsiChr6 100.00  24      0       0       1       24      29608223        29608200        1e-05   48.1    0       100     100
MY4F_Xa7QTL-200R        OsiChr6 100.00  18      0       0       1       18      29910098        29910115        0.017   36.2    0       100     100
MY4R_Xa7QTL-200R        OsiChr6 100.00  18      0       0       1       18      29910250        29910233        0.017   36.2    0       100     100

# blastn.Xa7Markershort-on-OsaDNA_Chr6.table
M5_Xa7QTL_BY    OsaChr6 94.35   336     18      1       152     486     26752440        26752775        6e-144    514   1       36      27
M5_Xa7QTL_BY    OsaChr6 93.44   259     15      1       245     503     27007346        27007090        2e-104    383   2       36      21
M5_56k_Xa7QTL_BY        OsaChr6 97.25   1746    23      13      1       1745    26776166        26777887        0.0      2935   25      100     100

RM20593F_Xa7QTL-118R    OsaChr6 100.00  25      0       0       1       25      26841939        26841963        3e-06   50.1    0       100     100
RM20576F_Xa7QTL-200L    OsaChr6 100.00  24      0       0       1       24      26625287        26625310        1e-05   48.1    0       100     100
RM20576R_Xa7QTL-200L    OsaChr6 100.00  24      0       0       1       24      26625576        26625553        1e-05   48.1    0       100     100
MY4F_Xa7QTL-200R        OsaChr6 100.00  18      0       0       1       18      26849409        26849426        0.016   36.2    0       100     100
MY4R_Xa7QTL-200R        OsaChr6 100.00  18      0       0       1       18      26849561        26849544        0.016   36.2    0       100     100

* 1. M5_M5-56K in rice japonica is chr6:27,965,437..28,022,337, Length=56,901 bp.
* 2. M5_M5-56K in rice japonica is redefined as chr6:27,965,001..28,023,000, Length=58,000 bp.
* 3. M5_M5-56K in rice indica is chr6:29,683,670..29,828,319, Length=144,650 bp.
* 4. M5_M5-56K in rice indica is redefined to chr6:29,683,001..29,829,000, Length=146,000 bp.
* 5. M5_M5-56K in rice aus Kasalath is chr6:26,752,440..26,776,166..26,777,887..(27,007,090), Length=25,448 (or 254,651) bp.
* 6. M5_M5-56K in rice aus Kasalath is redefined to chr6:26,625,001..27,007,500, Length=382,500 bp.

## run TRegGA using OsaXA7 as reference
sh runTRegGA_IRBB7-on-OsaXA7
sh runTRegGA_DV86-on-OsaXA7
sh runTRegGA_IRBB62-on-OsaXA7

\cp ${TRegGA_DIR}/run/Kasalath/Xa7MarkerDNA.fa .
QUERY=Xa7MarkerDNA.fa
\cp ${TRegGA_DIR}/run/Kasalath/Xa7Markershort.fa .
QUERYSHORT=Xa7Markershort.fa
OUTFMT='6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore gaps qcovs qcovhsp'
EVALUE="1e-100"
PERC_IDENTITY=92

## locate the markers in the assemblies
\cp ${TRegGA_DIR}/assembly/rfguided/${SYNONYM}-on-${TARGET_SEQNAME}/EVALUATION/${SYNONYM}-${TARGET_SEQNAME}.fa .
\cp ${TRegGA_DIR}/assembly/rfguided/${SYNONYM}-on-${TARGET_SEQNAME}/EVALUATION/${SYNONYM}-${TARGET_SEQNAME}.embl .
blastn -query ${QUERY} -subject ${SYNONYM}-OsaXA7.fa -evalue ${EVALUE} -perc_identity=${PERC_IDENTITY} -outfmt "${OUTFMT}" > blastn.Xa7MarkerDNA-on-${SYNONYM}-OsaXA7.table
blastn -query Xa7Markershort.fa -subject ${SYNONYM}-OsaXA7.fa -task "blastn-short" -word_size 11 -evalue 10 -perc_identity=100 -outfmt "${OUTFMT}" > blastn.Xa7Markershort-on-${SYNONYM}-OsaXA7.table


# IRBB7-OsaXA7.fa
M5_56k_Xa7QTL_BY        IRBB7-OsaXA7    99.14   813     5       1       1       813     157840  158650  0.0      1461   2       81      47
M5_56k_Xa7QTL_BY        IRBB7-OsaXA7    95.64   596     11      3       816     1410    158705  159286  0.0       942   15      81      34
RM20593F_Xa7QTL-118R    IRBB7-OsaXA7    100.00  25      0       0       1       25      228388  228412  5e-09   50.1    0       100     100
RM20576F_Xa7QTL-200L    IRBB7-OsaXA7    100.00  24      0       0       1       24      603     626     2e-08   48.1    0       100     100
RM20576R_Xa7QTL-200L    IRBB7-OsaXA7    100.00  24      0       0       1       24      995     972     2e-08   48.1    0       100     100
MY4F_Xa7QTL-200R        IRBB7-OsaXA7    100.00  18      0       0       1       18      236314  236331  4e-05   36.2    0       100     100
MY4R_Xa7QTL-200R        IRBB7-OsaXA7    100.00  18      0       0       1       18      236501  236484  4e-05   36.2    0       100     100

# DV86-OsaXA7.fa
M5_56k_Xa7QTL_BY        DV86-OsaXA7     98.96   673     5       1       1       673     156895  157565  0.0      1203   2       58      39
M5_56k_Xa7QTL_BY        DV86-OsaXA7     97.62   336     7       1       1038    1372    158018  158353  4e-165    575   1       58      19
RM20593F_Xa7QTL-118R    DV86-OsaXA7     100.00  25      0       0       1       25      223389  223413  5e-09   50.1    0       100     100
RM20576F_Xa7QTL-200L    DV86-OsaXA7     100.00  24      0       0       1       24      580     603     2e-08   48.1    0       100     100
MY4F_Xa7QTL-200R        DV86-OsaXA7     100.00  18      0       0       1       18      231077  231094  4e-05   36.2    0       100     100

# IRBB62-OsaXA7.fa
M5_56k_Xa7QTL_BY        IRBB62-OsaXA7   99.12   1025    7       1       1       1025    159837  160859  0.0      1842   2       59      59
RM20593F_Xa7QTL-118R    IRBB62-OsaXA7   100.00  25      0       0       1       25      230287  230311  5e-09   50.1    0       100     100
RM20576F_Xa7QTL-200L    IRBB62-OsaXA7   100.00  24      0       0       1       24      559     582     2e-08   48.1    0       100     100
MY4R_Xa7QTL-200R        IRBB62-OsaXA7   100.00  18      0       0       1       18      238209  238192  4e-05   36.2    0       100     100



##------------------------------------
## DJ123 scaffold_10 has VERY good coverage on the Xa7QTL region
## Transfer annotaion from rice aus reference Kasalath to rice aus DJ123 scaffold_10:197,410..199,218 
# DJ123 scaffold_10 full length is 1,279,957 bp; M5-56K region Length=1,809 bp.
cd ${TRegGA_DIR}/run
sh ${TRegGA_DIR}/bin/RATT_Ref-to-Assembly.sh

## run TRegGA using DJ123_scaffold10 as reference
cd ${TRegGA_DIR}/targets
\cp ${TRegGA_DIR}/run/ratt_DJ123_scaffold10-to-OsaXA7/EVALUATION/DJ123_scaffold10.fa DJ123_scaffold10.fasta
\cp ${TRegGA_DIR}/run/ratt_DJ123_scaffold10-to-OsaXA7/EVALUATION/DJ123_scaffold10.embl .
# for unknown reasons, the DJ123_scaffold10.gff3 is empty, so remake it from embl file:
seqret -sequence DJ123_scaffold10.embl -feature -snucleotide T -sformat1 embl -osformat2 gff3 -outseq DJ123_scaffold10.gff3

sh runTRegGA_IRBB7-on-DJ123_scaffold10
sh runTRegGA_DV86-on-DJ123_scaffold10
sh runTRegGA_IRBB62-on-DJ123_scaffold10

## Search for differentially expressed genes in the Xa7QTL region of DJ123_scaffold10
cd ${TRegGA_DIR}
sh ${TRegGA_DIR}/TRegGA_gth_RNAseqDE.sh

## Only two DE genes, OS06G0673800 and OS06G0676700, in all IRBB7, DV86 and IRBB62
# OS06G0673800: Conserved hypothetical protein
# OS06G0676700: Similar to High pI alpha-glucosidase

## Compare Talen binding site: the avrXa7 resistant plants should have the avrXa7 binding site at the DE genes
# IRBB7-DJ123_scaffold10.embl vs DJ123_scaffold10.embl
# DV86-on-DJ123_scaffold10.embl vs DJ123_scaffold10.embl
# IRBB62-on-DJ123_scaffold10.embl vs DJ123_scaffold10.embl

sh ${TRegGA_DIR}/0RECORD-talenXA7

## Find genomic positions of DE genes OS06G0673800 and OS06G0676700 in 
#  IRBB7-DJ123_scaffold10.embl, DV86-on-DJ123_scaffold10.embl, IRBB62-on-DJ123_scaffold10.embl
GTHFILE="gth.OsjPRT-on-DJ123_scaffold10 gth.OsjPRT-on-IRBB7-DJ123_scaffold10 gth.OsjPRT-on-DV86-DJ123_scaffold10 gth.OsjPRT-on-IRBB62-DJ123_scaffold10"
DEGENE="OS06T0673800 OS06T0676700"
cd ${TRegGA_DIR}/run/talen_avrXA7_on_DJ123_scaffold10
\rm -rf DEgenes_whereis_*
for k in ${GTHFILE}
do
for i in ${DEGENE}
do
grep "^PGS.*${i}" ${k} >> DEgenes_whereis_${k}
done
cat DEgenes_whereis_${k} >> DEgenes_whereis_all
done

## Compare the assemblies to the p53_P5L2_sequence_from_IRBB7_Xa7_containing_rice, 62 kb.
# OUTFMT is the tabular output format for BlastDB query
OUTFMT='6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore gaps qcovs qcovhsp'
EVALUE="1e-100"
PERC_IDENTITY=92

cd ${TRegGA_DIR}/run
mkdir blastn_p53P5L2
cd blastn_p53P5L2
cp /home/huangcy/MYWORK/Xa7QTL/Oryza_sativa_indica_IRBB7_Xa7QTL_p53P5L2/BY_p53P5L2_IRBB7_Xa7.fa .
\cp ${TRegGA_DIR}/run/talen_avrXA7_on_DJ123_scaffold10/*.fa .
\cp ${TRegGA_DIR}/run/talen_avrXA7_on_DJ123_scaffold10/*.embl .

QUERY=${TRegGA_DIR}/doc/BY_p53P5L2_IRBB7_Xa7.fa
QUERYNAME=p53P5L2_irbb7
QUERY=
QUERYNAME=

SUBJECT=DJ123_scaffold10.fa
SUBJECTNAME=DJ123_scaffold10
SUBJECT=IRBB7-DJ123_scaffold10.fa
SUBJECTNAME=IRBB7-DJ123_scaffold10
SUBJECT=DV86-DJ123_scaffold10.fa
SUBJECTNAME=DV86-DJ123_scaffold10
SUBJECT=IRBB62-DJ123_scaffold10.fa
SUBJECTNAME=IRBB62-DJ123_scaffold10

blastn -query ${QUERY} -subject ${SUBJECT} -evalue ${EVALUE} -perc_identity=${PERC_IDENTITY} -outfmt "${OUTFMT}" > blastn.${QUERYNAME}-on-${SUBJECTNAME}.table
blastn -query ${QUERY} -subject ${SUBJECT} -evalue ${EVALUE} -perc_identity=${PERC_IDENTITY} > blastn.${QUERYNAME}-on-${SUBJECTNAME}.aln

# BY_p53P5L2_IRBB7_Xa7 aligned 59,812bp (out of 62,882 bp, or 95.12%) with DJ123_scaffold10.
# BY_p53P5L2_IRBB7_Xa7 aligned 50,087bp (out of 62,882 bp, or 79.65%) with IRBB7-DJ123_scaffold10. 















## The only alternative way is to do RNA-Seq mapping on the assemblies regardless of the gene annotation, 
## as the real DE gene may be a noval gene and has not yet been annotated in rice japonica before.


```
